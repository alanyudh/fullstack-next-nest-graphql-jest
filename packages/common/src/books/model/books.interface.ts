export interface IBook {
  id: string;
  title: string;
  description: string;
  year: number;
}
