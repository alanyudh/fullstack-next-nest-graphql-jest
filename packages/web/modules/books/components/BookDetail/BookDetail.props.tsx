import { Book } from '../../models/book.model'

export interface BookDetailProps {
  book: Book
}
