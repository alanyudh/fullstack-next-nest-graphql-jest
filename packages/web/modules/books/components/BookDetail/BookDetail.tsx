import { Typography, Divider, Empty } from 'antd'
import { BookDetailProps } from './BookDetail.props'

const { Title, Paragraph, Text, Link } = Typography

const BookDetail: React.FC<BookDetailProps> = ({ book }: BookDetailProps) => {
  if (!book) return <Empty data-testid="book-detail-empty" />

  return (
    <Typography>
      <Title data-testid="book-detail-title">{book.title}</Title>
      <Paragraph data-testid="book-detail-desc">{book.description}</Paragraph>

      <Paragraph>
        <Text type="secondary" data-testid="book-detail-year">
          Published on {book.year}
        </Text>
      </Paragraph>

      <Divider />

      <Link href="/">back</Link>
    </Typography>
  )
}

export default BookDetail
