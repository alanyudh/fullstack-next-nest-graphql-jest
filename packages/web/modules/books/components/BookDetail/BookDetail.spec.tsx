import { Book } from '../../models/book.model'
import BookDetail from './BookDetail'
import { BookDetailProps } from './BookDetail.props'
import { render } from '../../../../test/testUtils'

describe('BookDetail', () => {
  let expectedProps: BookDetailProps

  beforeEach(() => {
    expectedProps = {
      book: new Book('1', 'title', 'description', 2009),
    }
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<BookDetail {...expectedProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render title, description, year', () => {
    const { getByText } = render(<BookDetail {...expectedProps} />)
    const title = getByText(expectedProps.book?.title)
    const description = getByText(expectedProps.book?.description)
    const year = getByText(/2009/)

    expect(title).toBeVisible()
    expect(description).toBeVisible()
    expect(year).toBeVisible()
  })

  it('should render empty information when there is no props', () => {
    expectedProps.book = null

    const { getByTestId } = render(<BookDetail {...expectedProps} />)
    const emptyBox = getByTestId('book-detail-empty')

    expect(emptyBox).toBeVisible()
  })
})
