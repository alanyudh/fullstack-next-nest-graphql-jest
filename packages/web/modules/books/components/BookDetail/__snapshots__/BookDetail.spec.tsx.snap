// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetail matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="ant-typography"
  >
    <h1
      class="ant-typography"
      data-testid="book-detail-title"
    >
      title
    </h1>
    <div
      class="ant-typography"
      data-testid="book-detail-desc"
    >
      description
    </div>
    <div
      class="ant-typography"
    >
      <span
        class="ant-typography ant-typography-secondary"
        data-testid="book-detail-year"
      >
        Published on 2009
      </span>
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <a
      class="ant-typography"
      href="/"
    >
      back
    </a>
  </article>
</DocumentFragment>
`;
